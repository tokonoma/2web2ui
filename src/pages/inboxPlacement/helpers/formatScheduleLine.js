import React from 'react';
import moment from 'moment';
import { STATUS, DURATION_HOURS } from '../constants/test';
import { FORMATS } from 'src/constants';
import { InlineItems } from 'src/components/structure';
import { Bold } from 'src/components/text';

const formatDateTime = start_time => moment(start_time).format(FORMATS.LONG_DATETIME_ALT);
const getHoursRemaining = start_time =>
  moment(start_time)
    .add(DURATION_HOURS, 'hours')
    .diff(moment(), 'hours');

export default function formatScheduleLine(status, start_time, end_time) {
  const formatted_start_time = formatDateTime(start_time);

  switch (status) {
    case STATUS.RUNNING: {
      return (
        <>
          <Bold>Sent</Bold> {formatted_start_time} and {getHoursRemaining(start_time)} hours
          remaining on test
        </>
      );
    }
    case STATUS.STOPPED: {
      return (
        <InlineItems compact>
          <span>
            <Bold>Sent</Bold> {formatted_start_time}
          </span>
          <span>
            <Bold>Stopped</Bold> {formatDateTime(end_time)}
          </span>
        </InlineItems>
      );
    }
    case STATUS.COMPLETED: {
      return (
        <InlineItems compact>
          <span>
            <Bold>Sent</Bold> {formatted_start_time}
          </span>
          <span>
            <Bold>Completed</Bold> {formatDateTime(end_time)}
          </span>
        </InlineItems>
      );
    }
    default: {
      return (
        <>
          <Bold>Sent</Bold> {formatted_start_time} and test status unknown
        </>
      );
    }
  }
}
