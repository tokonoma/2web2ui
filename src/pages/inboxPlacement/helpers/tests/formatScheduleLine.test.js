import formatScheduleLine from '../formatScheduleLine';
import moment from 'moment';

let formatLine;
let status;
let start_time;
let end_time;

describe('Promo Code async validator', () => {
  beforeEach(() => {
    const date = moment(new Date('2019-08-10T12:30:00-04:00'));
    Date.now = jest.fn(() => date);
    status = 'running';
    start_time = '2019-08-09T12:30:00-04:00';
    end_time = '2019-08-11T12:30:00-04:00';
    formatLine = formatScheduleLine;
  });

  it('should format start time and hours left for running test', () => {
    expect(formatLine(status, start_time, end_time)).toMatchSnapshot();
  });

  it('should format stopped test', () => {
    status = 'stopped';
    end_time = '2019-08-09T18:30:00-04:00';
    expect(formatLine(status, start_time, end_time)).toMatchSnapshot();
  });

  it('should format completed test', () => {
    status = 'completed';
    end_time = '2019-08-09T20:30:00-04:00';
    expect(formatLine(status, start_time, end_time)).toMatchSnapshot();
  });

  it('handle error state gracefully where status is unknown', () => {
    status = 'say_what?';
    expect(formatLine(status, start_time, end_time)).toMatchSnapshot();
  });
});
