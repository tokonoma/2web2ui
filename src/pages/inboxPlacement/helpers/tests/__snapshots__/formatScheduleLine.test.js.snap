// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Promo Code async validator handle error state gracefully where status is unknown 1`] = `
<React.Fragment>
  <Bold>
    Sent
  </Bold>
   
  Aug 09, 2019 at 12:30pm
   and test status unknown
</React.Fragment>
`;

exports[`Promo Code async validator should format completed test 1`] = `
<InlineItem
  compact={true}
>
  <span>
    <Bold>
      Sent
    </Bold>
     
    Aug 09, 2019 at 12:30pm
  </span>
  <span>
    <Bold>
      Completed
    </Bold>
     
    Aug 09, 2019 at 8:30pm
  </span>
</InlineItem>
`;

exports[`Promo Code async validator should format start time and hours left for running test 1`] = `
<React.Fragment>
  <Bold>
    Sent
  </Bold>
   
  Aug 09, 2019 at 12:30pm
   and 
  24
   hours remaining on test
</React.Fragment>
`;

exports[`Promo Code async validator should format stopped test 1`] = `
<InlineItem
  compact={true}
>
  <span>
    <Bold>
      Sent
    </Bold>
     
    Aug 09, 2019 at 12:30pm
  </span>
  <span>
    <Bold>
      Stopped
    </Bold>
     
    Aug 09, 2019 at 6:30pm
  </span>
</InlineItem>
`;
